doc/buffer.md

```diff
@@ -1,5 +1,147 @@
+You are reading a draft of the next documentation and it's in continuous update
+so if you don't find what you need please refer to:
+[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
+
 # Buffer
 
-You are reading a draft of the next documentation and it's in continuos update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
\ No newline at end of file
+The Buffer class creates a projection of raw data that can be consumed by
+script.
+
+## Methods
+
+### New
+
+Allocates a new Buffer object with a given length.
+
+```cpp
+static Buffer<T> New(napi_env env, size_t length);
+```
+
+- `[in] env`: The `napi_env` environment in which to create the Buffer object.
+- `[in] length`: The length of the buffer to allocate.
+
+Returns a new Buffer object.
+
+### New
+
+Wraps the provided `data` into a Buffer object.
+
+The Buffer object does not assume ownership for the data and expects it to be
+valid for the lifetime of the Buffer object. Since the Buffer is subject to
+garbage collection this overload is only suitable for data which is static and
+never needs to be freed.
```

Comment:
Can someone fact-check this?  I did some digging in the code and this appears to be true.

If so, it's a departure from NAN which would take ownership of the data and later free it.  Something to keep in mind for the conversion script, if someone was depending on this behavior there would be a memory leak in the converted code.

Reply:
I opened #258 to discuss